pinot-server/src/main/java/com/linkedin/pinot/server/starter/helix/SegmentOnlineOfflineStateModelFactory.java

```diff
@@ -89,26 +92,51 @@ public SegmentOnlineOfflineStateModel(String helixClusterName, String instanceId
 
     @Transition(from = "OFFLINE", to = "CONSUMING")
     public void onBecomeConsumingFromOnline(Message message, NotificationContext context) {
-      LOGGER.error("Unexpected transition from OFFLINE to CONSUMING for {}", message.getResourceName());
-      throw new RuntimeException("Unexpected state transition");
+      LOGGER.info("SegmentOnlineOfflineStateModel.onBecomeConsumingFromOffline() : " + message);
```

Comment:
check segment name here, and retain the unexpected transition for high-level consumer segments